Lots of 'EOF' messages after starting, but no data

Was looking for something like this. I have a (very old) raspberry pi near my smartmeter which I currently use to get the data to emoncms.
Now I'm moving several things over to Home Assistant and this is one of the last thing to do.
The only package running (at all) is the arm-one.
I configured the config.yml and ran the program.
It returns a lot of lines with '2023/12/08 13:55:43 EOF'. (Whether debug is enabled or not.)
When I hit any key the ssh session stops and I have to login again.
I would like to get this to work.
Can you help me out?

Kan ik achterhalen of 4 of 5 beter werkt met mijn meter?

Hij crasht nu met 
```
2023/12/08 20:14:23 Starting server...
2023/12/08 20:14:23 Listening on [::]:9988
2023/12/08 20:14:35 192.168.20.46:55498 connected
panic: runtime error: index out of range [0] with length 0

goroutine 18 [running]:
main.collectTelegrams(0x1580000, 0x141e1c0)
	/Users/chrisjansen/IdeaProjects/smartmeterBridge/bridge.go:192 +0x120
created by main.ReadRawTelegrams in goroutine 6
	/Users/chrisjansen/IdeaProjects/smartmeterBridge/bridge.go:167 +0x114
```

Die gaf de crash..

For future reference (dankzij jouw tip, voor mijn meter iig)
```
Kaifa	MA105	1-fase kleinverbruik	RJ11 6P6C	DSMR 4.0 (4.0.5 / 4.0.7)	115200 8N1	Ja	/KFM5
```
DSRM4 dus.
Gevonden op https://domoticx.com/p1-poort-slimme-meter-hardware/.

---edit
Een keer een herstart en weer gecrasht.
```
2023/12/08 20:28:23 EOF
panic: runtime error: index out of range [0] with length 0

goroutine 19 [running]:
main.collectTelegrams(0x502080, 0x440080)
	/Users/chrisjansen/IdeaProjects/smartmeterBridge/bridge.go:192 +0x120
created by main.ReadRawTelegrams in goroutine 5
	/Users/chrisjansen/IdeaProjects/smartmeterBridge/bridge.go:167 +0x114
```

--- a/bridge.go
+++ b/bridge.go
@@ -69,7 +69,9 @@ func getSerialConfig(port, dsmrVersion string) serial.Config {
 
 	confs["2.2"] = confs["2"]
 	confs["3"] = confs["2"]
-	confs["5"] = confs["4"]
+	conf5 := confs["4"]
+	conf5.StopBits = 1
+	confs["5"] = conf5
 	confs["5B"] = confs["5"]
 	confs["5L"] = confs["5"]
 	confs["5S"] = confs["5"]
@@ -174,12 +176,13 @@ func readLines(serialConfig serial.Config, rChan chan string) {
 	reader := bufio.NewReader(s)
 	for {
 		reply, err := reader.ReadString('\n')
-		if err != nil {
+		if err != nil && config.Verbose {
 			log.Println(err)
-			continue
 		}
 
-		rChan <- reply
+		if len(reply) > 0 {
+			rChan <- reply
+		}
 	}
 }
 

